test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedResultTests.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging.Testing;
+using Moq;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Core.Test
+{
+    public class AcceptedResultTests
+    {
+        public static TheoryData<object> ValuesData
+        {
+            get
+            {
+                return new TheoryData<object>
+                {
+                    null,
+                    "Test string",
+                    new ResourceStatusBody
+                    {
+                        EstimatedTime =DateTimeOffset.UtcNow.AddMinutes(5),
+                        Status = ResourceStatus.In_Progress,
+                    }
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ValuesData))]
+        public void AcceptedObjectResult_InitializesStatusCodeAndValue(object value)
+        {
+            // Arrange & Act
+            var result = new AcceptedResult("testlocation", value);
+
+            // Assert
+            Assert.Equal(StatusCodes.Status202Accepted, result.StatusCode);
+            Assert.Same(value, result.Value);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValuesData))]
+        public async Task AcceptedObjectResult_SetsStatusCodeAndValueAsync(object value)
+        {
+            // Arrange
+            var location = "/test/";
+            var httpContext = GetHttpContext();
+            var actionContext = GetActionContext(httpContext);
+            var result = new AcceptedResult(location, value);
+
+            // Act
+            await result.ExecuteResultAsync(actionContext);
+
+            // Assert
+            Assert.Equal(StatusCodes.Status202Accepted, httpContext.Response.StatusCode);
+            Assert.Same(value, result.Value);
+        }
+
+        [Fact]
+        public void AcceptedObjectResult_SetsLocation()
```

Comment:
Optionally you can remove this test. We usually don't test trivial property setting code (even if this existed in the other code).